src/torrent.cpp

```diff
@@ -9181,7 +9181,9 @@ namespace {
 		if (m_incomplete != 0xffffff) downloaders = m_incomplete;
 		else downloaders = m_peer_list ? m_peer_list->num_peers() - m_peer_list->num_seeds() : 0;
 
-		if (seeds == 0)
+		// If there is a single known seed and we're currently seeding then it's
+		// likely us, so consider the torrent otherwise seedless
+		if (seeds == 0 || (seeds == 1 && !is_paused()))
```

Comment:
there is one issue here. We can actually get here without being a seed, in which case `!is_paused()` cannot mean that we are the one seed.

on line 9146 we exit this function with 0 in case we aren't *finished* (i.e. we have downloaded all the files we want to download, but we may have de-selected some).

then we check `is_seed()`. I think this should probably take `is_seed()` into account as well.

It might be simpler to create a new variable, something like this:

```
int const self_seed = is_seed() && !is_paused() ? 1 : 0;
```

We can then compute `seeds` by subtracting this. Specifically, we really risk seeing ourself in the scrape data from the tracker, that's `m_complete`. If we happen to compute `seeds` by looking at our own peer list, we won't have ourself in there. So, rather than putting this logic here, I think it would be more accurate to change line 9178 to be:

```
if (m_complete != 0xffffff) seeds = std::min(0, m_complete - self_seed);
```

And move the comment there too.

Would that work?

Reply:
That looks good, though should that `min` be a `max`? Otherwise we'd never end up with more than 0 seeds.

Reply:
heh, yes :)